### ISSUE TITLE
MessageFormatter generator creates 422 error with HttpStatusPush reporter

### ISSUE THREAD

**Issue by joequant** [buildbot/buildbot#5670]
The code below works but when I change the message format object to MessageFormatter
I get a 422 error.  Would like some pointers as to what could be causing the issue.

```
def getresults(props):
    return {"body": ("failed %s" % repr(props))}

script_dir = os.path.dirname(os.path.realpath(__file__))
token = open(os.path.join(script_dir, "token"), 'r').read().strip()

from buildbot.reporters.generators.build import BuildStatusGenerator
from buildbot.reporters.message import DeprecatedMessageFormatterBuildJson
generator = BuildStatusGenerator(
    subject="Build failed",
    mode="failing",
    message_formatter=DeprecatedMessageFormatterBuildJson(getresults)
)

mn = reporters.HttpStatusPush(
    "https://api.github.com/repos/joequant/key4hep-container/issues/1/comments",
    auth=("joequant", token,),
    generators=[generator])

```

**Comment by p12tic** [comment]
@joequant Thanks a lot for the bug report, I'll look into this ASAP.

**Comment by p12tic** [comment]
@joequant At this moment there's no proper migration path for HttpStatusPush to MessageFormatters. I was too eager to enable deprecation warnings for all code paths that will eventually be removed, but in this case this is not the time for that yet.

So you don't need to migrate to the MessageFormatter for the HttpStatusPush yet.

It's likely that by Buildbot 2.10 the warnings will be enabled again, but by then there will be a proper solution.

**Comment by p12tic** [comment]
I'm closing this issue as I think it's resolved in 2.9.2. Please reopen the issue if you think I'm wrong and misunderstood the bug or if deprecation warnings are still being thrown for old usage of `HttpStatusPush`. Thanks you!
